### PR TITLE
BUG: loc.setitem corner case

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

cc @phofl if you've got any thoughts on how to handle this case

### CODE REVIEW

File: pandas/core/indexing.py [hunk 1]
```diff
@@ -1670,6 +1671,12 @@ def _setitem_with_indexer_split_path(self, indexer, value):
                 # We are trying to set N values into M entries of a single
                 #  column, which is invalid for N != M
                 # Exclude zero-len for e.g. boolean masking that is all-false
+
+                if len(value) == 1 and not is_integer(info_axis):
+                    # This is a case like df.iloc[:3, [1]] = [0]
+                    #  where we treat as df.iloc[:3, 1] = 0
+                    return self._setitem_with_indexer((pi, info_axis[0]), value[0])
```
Comment (phofl) [inline pandas/core/indexing.py:1678]: Is there a reason for not using ``_setitem_single_column``? Seems to work when trying out locally. LGTM otherwise

Comment (jbrockmendel) [reply pandas/core/indexing.py:1678]: either way works.  this seemed like a more obvious "we are reducing this to a simpler case" representation

Comment (phofl) [reply pandas/core/indexing.py:1678]: Ok, this is fine with me then.

File: pandas/tests/indexing/multiindex/test_setitem.py [hunk 2]
```diff
@@ -210,6 +210,11 @@ def test_multiindex_assignment(self):
         with pytest.raises(ValueError, match=msg):
             df.loc[4, "c"] = [0]
 
+        # But with a length-1 listlike column indexer this works
+        df.loc[4, ["c"]] = [0]
```
Comment (jreback) [inline pandas/tests/indexing/multiindex/test_setitem.py:214]: what happens with this case now?

**Review by jreback — CHANGES_REQUESTED** [review]
also merge master

File: pandas/tests/indexing/multiindex/test_setitem.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/indexing/multiindex/test_setitem.py:215]: behaves like `df.loc[4, "c"] = 0`, just edited the comment to make that clear

**Comment by jbrockmendel** [comment]
rebased+green

**Review by jreback — APPROVED** [review]

**Pull request merged**